docs/reference/search/async-search.asciidoc

```diff
@@ -0,0 +1,200 @@
+parameters and request body as the <<search-search,search API>>.
+
+[source,console,id=submit-async-search-date-histogram-example]
+--------------------------------------------------
+POST /sales*/_async_search?size=0
+{
+    "sort" : [
+      { "date" : {"order" : "asc"} }
+    ],
+    "aggs" : {
+        "sale_date" : {
+             "date_histogram" : {
+                 "field" : "date",
+                 "calendar_interval": "1d"
+             }
+         }
+    }
+}
+--------------------------------------------------
+// TEST[setup:sales]
+// TEST[s/size=0/size=0&wait_for_completion=0/]
+
+The response contains an identifier of the search being executed.
+You can use this ID to later retrieve the search's final results.
+The currently available search
+results are returned as part of the <<search-api-response-body,`response`>> object.
+
+[source,console-result]
+--------------------------------------------------
+{
+  "id" : "FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=", <1>
+  "version" : 0,
+  "is_partial" : true, <2>
+  "is_running" : true, <3>
+  "start_time_in_millis" : 1583945890986,
+  "expiration_time_in_millis" : 1584377890986,
+  "response" : {
+    "took" : 1122,
+    "timed_out" : false,
+    "num_reduce_phases" : 0,
+    "_shards" : {
+      "total" : 562, <4>
+      "successful" : 3, <5>
+      "skipped" : 0,
+      "failed" : 0
+    },
+    "hits" : {
+      "total" : {
+        "value" : 157483, <6>
+        "relation" : "gte"
+      },
+      "max_score" : null,
+      "hits" : [ ]
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=/$body.id/]
+// TESTRESPONSE[s/1583945890986/$body.start_time_in_millis/]
+// TESTRESPONSE[s/1584377890986/$body.expiration_time_in_millis/]
+// TESTRESPONSE[s/"took" : 1122/"took": $body.response.took/]
+// TESTRESPONSE[s/"total" : 562/"total": $body.response._shards.total/]
+// TESTRESPONSE[s/"successful" : 3/"successful": $body.response._shards.successful/]
+// TESTRESPONSE[s/"value" : 157483/"value": $body.response.hits.total.value/]
+
+<1> Identifier of the async search that can be used to monitor its progress, retrieve its results, and/or delete it.
+<2> Whether the returned search results are partial or final
+<3> Whether the search is still being executed or it has completed
+<4> How many shards the search will be executed on, overall
+<5> How many shards have successfully completed the search
+<6> How many documents are currently matching the query, which belong to the shards that have already completed the search
+
+It is possible to block and wait until the search is completed up to a certain
+timeout by providing the `wait_for_completion` parameter, which defaults to
+`1` second.
+
+The submit async search API supports the same <<search-search-api-query-params,parameters>>
+as the search API, though some have different default values:
```

Comment:
I think we can list the ones that can be changed and omit the other. `batched_reduce_size`, `request_cache` and `max_concurrent_shard_requests` + the specific options (wait_for_completion, keep_alive) should be enough for this API.  I think we said that we want  to remove `pre_filter_shard_size` so I'd remove it from here, same for `ccs_minimize_roundtrips` which sounds like it could be changed but cannot.
So to be clear, a concrete list that presents all these options and then we can add a link to the search source documentation that should be unchanged from normal search request ?

Reply:
`batched_reduce_size` is also important to document since that's the granularity at which partial results will be made available.

Reply:
agreed, I was planning on looking back to this once we disallow setting pre_filter_shard_size and maybe others, but I can address the docs in the meantime.

Reply:
> we can add a link to the search source documentation 

there is already a link. Why document max_concurrent_shard_request here? The default is the same as for search?

Reply:
because I thought we would have the exhaustive list of options here rather than linking to the entire search request option. We only accept a subset of it so better to be explicit ?

Reply:
the thing is that there's a ton of options that can be set to the search request. See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-search-api-query-params . I thought it's better to just link to search rather than duplicate its docs, and mention the special cases or important aspects only in the async search docs.  Would you rather list all of the search options also here?

Reply:
Ok for a simple link but we should at least document `batched_reduce_size` explicitly here for the reason exposed above. We can also add the list of options that are not accepted here (`scroll`, `ccs_minimize_roundtrips` and `pre_filter_shard_size`)  ?

Reply:
yep that's what I had in mind. batched_reduce_size makes a lot of sense to explain, I already did, I will push shortly